lifecycle-operator/controllers/common/eventsender.go

```diff
@@ -15,28 +20,100 @@ type IEvent interface {
 
 // ===== Main =====
 
-func NewEventSender(recorder record.EventRecorder) IEvent {
-	return newK8sSender(recorder)
+type EventMultiplexer struct {
+	logger   logr.Logger
+	emitters []IEvent
+}
+
+func NewEventMultiplexer(logger logr.Logger, recorder record.EventRecorder, client ce.Client) *EventMultiplexer {
+	multiplexer := &EventMultiplexer{
+		logger: logger,
+	}
+	multiplexer.register(newCloudEventSender(logger, client))
+	multiplexer.register(NewK8sSender(recorder))
+	return multiplexer
+}
+
+func (e *EventMultiplexer) register(emitter IEvent) {
+	if emitter != nil {
+		e.emitters = append(e.emitters, emitter)
+	}
+}
+
+func (e *EventMultiplexer) Emit(phase apicommon.KeptnPhaseType, eventType string, reconcileObject client.Object, status string, message string, version string) {
+	for _, emitter := range e.emitters {
+		e.logger.Info(fmt.Sprintf("Emitting event using %T", emitter))
+		emitter.Emit(phase, eventType, reconcileObject, status, message, version)
+	}
 }
 
 // ===== Cloud Event Sender =====
-// TODO: implement Cloud Event logic
+
+type cloudEvent struct {
+	client ce.Client
+	logger logr.Logger
+}
+
+func newCloudEventSender(logger logr.Logger, client ce.Client) *cloudEvent {
```

Comment:
because I want to forbid the direct usage of Cloud Events. We should always expose K8s events and only when the CE endpoint is configured **also** Cloud Events. If we expose this, somewhere in the code we could bind only CE bypassing k8s events.